vectorizers/linear_optimal_transport.py

```diff
@@ -1565,6 +1565,42 @@ def fit(
                 cachedir=self.cachedir,
             )
 
+        elif isinstance(X, GeneratorType) or isinstance(vectors, GeneratorType):
+            if reference_vectors is None:
+                raise ValueError("WassersteinVectorizer on a generator must specify reference_vectors!")
+            assert reference_vectors.shape[0] == self.reference_size
+
+            if n_distributions is None:
+                raise ValueError("WassersteinVectorizer on a generator must specify "
+                                 "how many distributions are to be vectorized!")
+
+            if vector_dim is None:
+                vector_dim = 1024 # Guess a largeish dimension and hope for the best
```

Comment:
# Guess a largeish dimension and hope for the best
This comment made my day. :-)